pkg/kademlia/kademlia.go

```diff
@@ -180,6 +172,13 @@ func (k *Kademlia) lookup(ctx context.Context, target dht.NodeID, opts discovery
 		return err
 	}
 
+	if opts.bootstrap {
+		for _, v := range opts.bootstrapNodes {
+			n := v
```

Comment:
why is a copy necessary here?

Reply:
opts.bootstrapNodes is type `[]proto.Node` and `newPeerDiscovery` wants it to be type `[]*proto.Node`

Reply:
You should be able to use `&v`.

Reply:
ok, I couldn't remember if it reuses the address space there. I now understand @jenlij 's comment.

Reply:
done